### ISSUE TITLE
undefined method `setup' for Config:Module (NoMethodError)      in Rails 5.0.0.1, Ruby 2.3.4

### ISSUE THREAD

**Issue by 100010** [holli/auto_strip_attributes#22]
when I tried to implement method which removes full space like

myapp/config/initializers/auto_strip_attributes.rb

```ruby
AutoStripAttributes::Config.setup do
  set_filter strip_full_width_space: false do |value|
    unless value.nil? || !value.is_a?(String)
      value.remove(/\A[\s　]+|[\s　]+\z/)
    end
  end
end
```

then  ` $ bundle exec rails s` or  ` $ bundle exec rails c`

myapp/config/initializers/auto_strip_attributes.rb:1:in `<top (required)>': undefined method `setup' for Config:Module (NoMethodError)

**Comment by holli** [comment]
Hmm weird. I tried with (Ruby 2.3.1, Rails 4.2.8) & (Ruby 2.4.0, Rails 5.1.1) and I couldn't reproduce this error.

Have you tried with different combinations and does it work then?

Have you done something else that could influence loading of the gem?

Your error says `for Config:Module (NoMethodError)` which is also weird because that one is defined as a class in the gem. Have you defined it as a module in somewhere in your own code?

**Comment by yashasvj** [comment]
I still have this issue. Below is the TraceBack
`
Traceback (most recent call last):
	45: from bin/rails:14:in `<main>'
	44: from bin/rails:14:in `require'
	43: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/engine/commands.rb:7:in `<top (required)>'
	42: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/engine/commands.rb:7:in `require'
	41: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/commands.rb:16:in `<top (required)>'
	40: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/command.rb:44:in `invoke'
	39: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/command/base.rb:63:in `perform'
	38: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	37: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	36: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	35: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/commands/console/console_command.rb:96:in `perform'
	34: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/command/actions.rb:16:in `require_application_and_environment!'
	33: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/application.rb:329:in `require_environment!'
	32: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/application.rb:329:in `require'
	31: from /Users/yash/Projects/FINALCAD/rails_engine/outpost/spec/dummy/config/environment.rb:5:in `<top (required)>'
	30: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/application.rb:353:in `initialize!'
	29: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/initializable.rb:58:in `run_initializers'
	28: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:205:in `tsort_each'
	27: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:226:in `tsort_each'
	26: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:347:in `each_strongly_connected_component'
	25: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:347:in `call'
	24: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:347:in `each'
	23: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:349:in `block in each_strongly_connected_component'
	22: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:415:in `each_strongly_connected_component_from'
	21: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:415:in `call'
	20: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/initializable.rb:48:in `tsort_each_child'
	19: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/initializable.rb:48:in `each'
	18: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:421:in `block in each_strongly_connected_component_from'
	17: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:431:in `each_strongly_connected_component_from'
	16: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:422:in `block (2 levels) in each_strongly_connected_component_from'
	15: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	14: from /Users/yash/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/tsort.rb:228:in `block in tsort_each'
	13: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/initializable.rb:59:in `block in run_initializers'
	12: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/initializable.rb:30:in `run'
	11: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/initializable.rb:30:in `instance_exec'
	10: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/engine.rb:611:in `block in <class:Engine>'
	 9: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/engine.rb:611:in `each'
	 8: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/engine.rb:612:in `block (2 levels) in <class:Engine>'
	 7: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/engine.rb:654:in `load_config_initializer'
	 6: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/activesupport-5.1.4/lib/active_support/notifications.rb:168:in `instrument'
	 5: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/railties-5.1.4/lib/rails/engine.rb:655:in `block in load_config_initializer'
	 4: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:286:in `load'
	 3: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:258:in `load_dependency'
	 2: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:286:in `block in load'
	 1: from /Users/yash/.rvm/gems/ruby-2.5.0@outpost/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:286:in `load'
/Users/yash/Projects/FINALCAD/rails_engine/outpost/config/initializers/config.rb:1:in `<top (required)>': undefined method `setup' for Config:Module (NoMethodError)
`

**Comment by holli** [comment]
Original issue might be fixed in gem version 2.4.

@yashasvj  you are using plain `Config.setup` . That won't work anyways. It's supposed to be `AutoStripAttributes::Config.setup`

I'm closing this issue now. If its still a problem please open again.

- you could check that the gem is in gemfile as it supposed (so its required automatically)
- also check if the gem is required at anypoint, so e.g. `puts AutoSripAttributes` in that file, is there something weird in the output.
